### PR TITLE
feat(cb2-4705): call put api with new test result object

### PR DESCRIPTION
**Pull request by mcadenas-bjss**
## Call PUT with new object created from section forms

[link to ticket number](https://dvsa.atlassian.net/browse/CB2-4705)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by andrewf-bjss — DISMISSED** [review]

**Review by kpearmanbjss — DISMISSED** [review]

**Review by gjulien-bjss — COMMENTED** [review]
Make sure the `package-lock.json` will run `npm audit` when using node 14

**Review by gjulien-bjss — APPROVED** [review]

**Review by tahayazici — APPROVED** [review]
Nice one!

**Review by andrewf-bjss — DISMISSED** [review]

**Review by kpearmanbjss — DISMISSED** [review]

**Review by tahayazici — APPROVED** [review]

File: src/app/forms/components/defect/defect.component.html [hunk 1]
```diff
@@ -1,46 +1,52 @@
...
+        data: combined(getControlValue('deficiencyId'), getControlValue('deficiencySubId'), getControlValue('deficiencyText'))
+      }"
     ></ng-template>
 
-    <ng-template
+    <!-- <ng-template
       [ngTemplateOutlet]="viewDefectRow"
       [ngTemplateOutletContext]="{ label: 'Location', data: combined(mapLocationText(data.additionalInformation?.location)) }"
-    ></ng-template>
+    ></ng-template> -->
```
Comment (andrewf-bjss) [inline src/app/forms/components/defect/defect.component.html:34]: Commented out, but might still be needed...  (same at lines 46-49?)

File: src/app/forms/components/defects/defects.component.ts [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+    return this.form.get(['testTypes', '0', 'defects']) as CustomFormArray;
+  }
+
+  getDefectForm(i: number) {
+    return this.defectsForm.controls[i] as CustomFormGroup;
+  }
 
-  constructor() {
-    this.forms = new Array(this.defects?.length).fill(null);
+  trackDefects(index: number, defect: AbstractControl): number {
```
Comment (andrewf-bjss) [inline src/app/forms/components/defects/defects.component.ts:23]: Nitpick - this is called `trackByFn` in other components...

```suggestion
  trackByFn(index: number, defect: AbstractControl): number {
```

**Review by mcadenas-bjss — COMMENTED** [review]
This would have been two separate PRs but ended up pushing CB2-4734 work into this one. Given that it's dependent on this work, I've left it in.
The additional work aims to fix how custom section such as `defects` are rendered as part of the master template.

**Review by andrewf-bjss — APPROVED** [review]

**Review by kpearmanbjss — DISMISSED** [review]

**Review by kpearmanbjss — APPROVED** [review]

**Review by andrewf-bjss — APPROVED** [review]

**Review by kpearmanbjss — APPROVED** [review]

**Review by andrewf-bjss — APPROVED** [review]

**Pull request merged**